nomad/container.py

```diff
@@ -165,18 +167,28 @@ def is_stopped(self):
         return self._container.status_code == constants.CONTAINER_STOPPED
 
     @property
+    def lxd_name(self):
+        """ Returns the name of the container that is used in the scope of LXD.
+
+        This name id supposed to be unique among all the containers managed by LXD.
+        """
+        # Note: all container names must be a valid hostname! That is: maximum 63 characters, no
+        # dots, no digit at first position, made entirely of letters/digits/hyphens, ...
+        if not hasattr(self, '_lxd_name'):
+            lxd_name_prefix = '{project_name}-{name}'.format(
```

Comment:
In single-container projects (the vast majority of use cases), we end up with duplicated project names (example: `foo-foo-20504857943`) in the LXD container names because of the way the config is passed around at instantiation.

Wouldn't it be nice if we could tidy this up? Suggestion:

1. Get rid of the creation counter logic, make `name` a required argument.
2. Assign auto-generated based on `enumerate()` at https://github.com/lxd-nomad/lxd-nomad/blob/improve_multi_containers_names/nomad/conf/config.py#L61
3. At https://github.com/lxd-nomad/lxd-nomad/blob/improve_multi_containers_names/nomad/conf/config.py#L66 override name so it's empty.
4. In this very line, don't include `name` if it's empty
5. Use `lxd_name` everywhere in our log messages.